File: OJ/Leetcode/Algorithm/581. Shortest Unsorted Continuous Subarray.py
class Solution(object):
    def findUnsortedSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        l = 0
        r = n-1
        ans = 0
        while l < n-1 and nums[l] <= nums[l+1]:
            l += 1
        while r >= 1 and nums[r-1] <= nums[r]:
            r -= 1
        
        if l < r:
            mi = min(nums[l:r+1])
            ma = max(nums[l:r+1])
            while l >= 1 and nums[l-1] > mi:
                l -= 1
            while r < n-1 and nums[r+1] < ma:
                r += 1
            ans = r-l+1
        return ans
        

File: OJ/Leetcode/Algorithm/581. Shortest Unsorted Continuous Subarray.cpp
public int findUnsortedSubarray(int[] nums) {
    int i = 0, j = -1, max = Integer.MIN_VALUE, min = Integer.MAX_VALUE;
    
    for (int l = 0, r = nums.length - 1; r >= 0; l++, r--) {
        max = Math.max(max, nums[l]);
        if (nums[l] != max) j = l;
        
        min = Math.min(min, nums[r]);
        if (nums[r] != min) i = r;
    }
    
    return (j - i + 1);
}
